routes/web/project/billing.rb

```diff
@@ -69,9 +69,30 @@ class CloverWeb
         r.redirect @project.path + "/billing"
       end
 
+      # Pre-authorize card to check if it is valid, if so
+      # authorization won't be captured and will be refunded immediately
+      customer_stripe_id = setup_intent["customer"]
+      payment_intent = Stripe::PaymentIntent.create({
+        amount: 100, # 100 cents to charge $1.00
+        currency: "usd",
+        confirm: true,
+        off_session: true,
+        capture_method: "manual",
+        customer: customer_stripe_id,
+        payment_method: stripe_id
+      })
+
+      if payment_intent.status != "requires_capture"
+        Clog.emit("Couldn't pre-authorize card") { {card_authorization: {project_id: @project.id, customer_stripe_id: customer_stripe_id, status: payment_intent.status}} }
```

Comment:
Should we delete the customer and payment method from Stripe?

Reply:
keeping it intentionally to check the details if we can't authorize the card